### ISSUE TITLE
when i produce msgs to an unknow host,ti don't report any error.

### ISSUE THREAD

**Issue by ShmSeeker** [segmentio/kafka-go#391]
**Describe the bug**
when i produce msgs to an unknow host,ti don't report any error.
**Kafka Version**
What version(s) of Kafka are you testing against?
2.3.1
**To Reproduce**
Steps to reproduce the behavior.  Bonus points for a code sample.
```go
	brokers := strings.Split(hosts,";")
	producer := kafka.NewWriter(kafka.WriterConfig{
		Brokers: brokers,
		Topic:   topic,
		QueueCapacity: queueCap,
		BatchBytes: batchBytes,
		BatchSize:  batchSize,
		BatchTimeout:  time.Duration(batchTimeout)*time.Millisecond,
		Async:	true,
		RequiredAcks: 1,
		CompressionCodec: lz4.CompressionCodec{},
		Balancer: kafka.Murmur2Balancer{},
	})

	producedMsg := kafka.Message{
		Value: msg,
	}

	fmt.Println("start to produce msgs.")
	start := time.Now()
	for i := 0; i < produceNum ; i++ {
		err := producer.WriteMessages(context.Background(), producedMsg)
		if err != nil {
			fmt.Printf("Produce %d msg falied.",i)
		}
	}
```
**Expected behavior**

**Additional context**

**Comment by achille-roussel** [comment]
Hello @ShmSeeker, thanks for reporting.

Your producer is configured with `Async: true`, in this mode messages are published in a "fire and forget" fashion, so errors are ignored.

Could you try without this setting and confirm whether you are getting errors reported?

**Comment by ShmSeeker** [comment]
producer := kafka.NewWriter(kafka.WriterConfig{
		Brokers: "localhost:9093",
		Topic:   topic,
		QueueCapacity: queueCap,
		BatchBytes: batchBytes,
		BatchSize:  batchSize,
		BatchTimeout:  time.Duration(batchTimeout)*time.Millisecond,
		Async:	false,
		RequiredAcks: 1,
		CompressionCodec: lz4.CompressionCodec{},
		Balancer: kafka.Murmur2Balancer{},
	})


it's send msg success........!!
start to produce msgs.
Produce msgs falied.producer 500000 msgs cost 11.861636 s 
Qps is 4.215270 w/s

**Comment by ShmSeeker** [comment]
when i try to subscribe the topic, it report: the compression code is invalid or its codec has not been imported.
but i cann't find optionnal parm when create reader.

`consumer := kafka.NewReader(kafka.ReaderConfig{
		Brokers:  	strings.Split(hosts,";"),
		GroupID:   	topic,
		Topic:     	topic,
		MinBytes:  	1, // 10KB
	})

	fmt.Println("Begin to consume topic. ")
	for {
		m, err := consumer.ReadMessage(context.Background())
		if err != nil {
			fmt.Printf("consume falied, err is %s\n ", err)
			break
		}
		fmt.Printf("message at offset %d: %s = %s\n", m.Offset, string(m.Key), string(m.Value))
	}
`

**Comment by achille-roussel** [comment]
This was solved by v0.4, closing.
